### PR TITLE
Optimize CI a bit

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -72,57 +64,39 @@ jobs:
...
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.java }}
-
       - name: Build and test
         if: startsWith(matrix.os, 'windows') == true
-        run: ./gradlew.bat build shadowJar proguard
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build shadowJar proguard
```
Comment (Goooler) [inline .github/workflows/build.yml:76]: https://github.com/gradle/gradle-build-action

Ref https://github.com/detekt/detekt/pull/3723, it's really so fast to run jobs, and many other advantages.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -69,86 +61,61 @@ jobs:
       fail-fast: true
       matrix:
         os: [ ubuntu-latest, macOS-latest, windows-latest ]
-        java: [ 8, 9, 10, 11, 12, 13, 14, 15, 16 ]
-
+        java: [ 8, 9, 10, 11, 12, 13, 14, 15, 16, 17 ]
```
Comment (Goooler) [inline .github/workflows/build.yml:64]: Run tests on java 17.

Comment (Goooler) [reply .github/workflows/build.yml:64]:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':brut.apktool:apktool-cli:proguard'.
> java.io.IOException: Can't read [/opt/hostedtoolcache/Java_Zulu_jdk/17.0.2-8/x64/jmods/java.base.jmod(;;;;;;;!**.jar;!module-info.class)] (Can't process class [jdk/internal/loader/URLClassPath$JarLoader.class] (Unsupported version number [61.0] (maximum 60.65535, Java 16)))
```

**Comment by iBotPeaches** [comment]
I didn't even know `v3` of official actions came out. I have some reading to do, then will be back to this review.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -70,85 +62,54 @@ jobs:
...
           distribution: 'zulu'
           java-version: ${{ matrix.java }}
-
       - name: Build and test
-        if: startsWith(matrix.os, 'windows') == true
-        run: ./gradlew.bat build shadowJar proguard
-
-      - name: Build and test
-        if: startsWith(matrix.os, 'windows') != true
-        run: ./gradlew build shadowJar proguard
```
Comment (Goooler) [inline .github/workflows/build.yml:97]: Seems this 2 same logic can be merged now.

**Review by iBotPeaches — APPROVED** [review]

**Pull request merged**